internal/pkg/agent/vault/vault_linux.go

```diff
@@ -0,0 +1,143 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build linux
+// +build linux
+
+package vault
+
+import (
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"golang.org/x/crypto/pbkdf2"
+)
+
+const saltSize = 8
+
+type Vault struct {
+	path string
+	key  []byte
+}
+
+// Open initializes the vault store
+func New(path string) (*Vault, error) {
+	dir := filepath.Dir(path)
+
+	// If there is no specific path then get the executable directory
+	if dir == "." {
```

Comment:
Don't we have something in path package to give us that path?

Reply:
from the source
https://github.com/golang/go/blob/19309779ac5e2f5a2fd3cbb34421dafb2855ac21/src/path/filepath/path.go#L580
```
// If the path is empty, Dir returns ".".
```